File: src/test/java/com/pepebot/testing/IntegrationTestBase.java
package com.pepebot.testing;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.test.context.ActiveProfiles;

@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
@ActiveProfiles("integration")
public abstract class IntegrationTestBase {

	@Autowired
	protected TelegramProperties telegramProperties;
}

File: src/test/kotlin/com/pepebot/testing/IntegrationTestBase.kt
package com.pepebot.testing

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.test.context.ActiveProfiles

@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
@ActiveProfiles("integration")
abstract class IntegrationTestBase {
    @Autowired
    protected lateinit var telegramProperties: TelegramProperties
}
